cogs/nodemanager.py

```diff
@@ -0,0 +1,153 @@
+# Discord Packages
+import discord
+import lavalink
+from discord.ext import commands
+from lavalink import Node
+
+import codecs
+
+import yaml
+
+from .utils.mixplayer import MixPlayer
+
+
+class NodeManager(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.settings = self.bot.settings
+        self.logger = self.bot.main_logger.bot_logger.getChild("NodeManager")
+
+        music_extensions = [
+            'cogs.music',
+            'cogs.musicevents'
+        ]
+
+        if not hasattr(bot, 'lavalink'):
+            bot.lavalink = lavalink.Client(bot.user.id, player=MixPlayer)
+
+            self.load_nodes_from_file()
+
+            bot.add_listener(bot.lavalink.voice_update_handler, 'on_socket_response')
+
+            for extension in music_extensions:
+                try:
+                    self.logger.debug("Loading extension %s" % extension)
+                    self.bot.load_extension(extension)
+                except Exception:
+                    self.logger.exception("Loading of extension %s failed" % extension)
+
+    def load_nodes_from_file(self):
+        with codecs.open(f"{self.bot.datadir}/config.yaml", 'r', encoding='utf8') as f:
+            conf = yaml.load(f, Loader=yaml.SafeLoader)
+
+            name_cache = []
+            new_nodes = []
+
+            for node in self.bot.lavalink.node_manager.nodes:
+                name_cache.append(node.name)
+
+            for node in conf['lavalink nodes']:
+                if conf['lavalink nodes'][node]['name'] in name_cache:
```

Comment:
This does not work for the current node format presented in data/config.yaml.example
(because the format I made is stupid).